### PR TITLE
fix some env vars, convert ssh urls

### PR DESCRIPTION
**Pull request by sabrinabrogren**
PR checklist:

- [x] Documentation is up-to-date
- [x] `changelog.d/<issue>.<type>` is a file with the _what_, _why_, and _how_ of the change.
  - \<issue> is `pa-312` (Linear ticket), `gh-1234` (GitHub issue), or `new-gizmo` (unique semantic name)
  - \<type> is `added`, `changed`, `fixed`, or `infra`.
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure on any of this, please see the [Contribution guidelines](https://semgrep.dev/docs/contributing/how-to-contribute/)!

Fixes CLI-267

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -21,6 +21,7 @@ This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html
 - Error messages in JSON output used to contain ANSI color codes,
   this was fixed so they are now uncolored (and readable!) plain text.
 - Fixed config parsing bug to cut CI scan speeds down significantly
+- Fixed parsing of CI environment variables for Circle CI, Bitbucket, and Buildkite
```
Comment (underyx) [inline CHANGELOG.md]: This is the changelog of last week's release — could you add this entry as a file on changelog.d/ instead?

CleanShot 2022-07-12 at 13 25 40@2x

File: cli/src/semgrep/meta.py [hunk 2]
```diff
@@ -22,6 +22,35 @@
 logger = getLogger(__name__)
 
 
+def get_url_from_ssh_url(ssh_url: Optional[str]) -> Optional[str]:
+    """Gets regular url from ssh_url"""
+    if not ssh_url or not ssh_url.startswith("git@") or not ssh_url.endswith(".git"):
+        return ssh_url
+
+    return f"https://{ssh_url[4:-4]}"
```
Comment (underyx) [inline cli/src/semgrep/meta.py]: there are a lot of possible git URL formats, might be worth checking if this function can be made more generic to normalize more of them https://stackoverflow.com/q/31801271

File: changelog.d/cli-267.fixed [hunk 3]
```diff
@@ -0,0 +1 @@
+Fixed parsing of CI environment variables for Circle CI, Bitbucket, and Buildkite
```
Comment (underyx) [inline changelog.d/cli-267.fixed]:
```suggestion
Semgrep App's links to repositories, code snippets, and pull requests are no longer broken for projects that run Semgrep in Circle CI, Bitbucket, and Buildkite.
```

Comment (underyx) [reply changelog.d/cli-267.fixed]: ^intends to be more user-centric as opposed to code change centric

**Review by underyx — APPROVED** [review]

File: cli/src/semgrep/meta.py (hunk 2 above)
Comment (sabrinabrogren) [reply cli/src/semgrep/meta.py]: Ok I rewrote it to cover everything except the local filesystem transport protocol!

**Pull request merged**